timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -181,7 +181,8 @@ def iter_items(self) -> Iterable[Any]:
     def num_items(self):
         return len(self.item_ids)
 
-    def num_timesteps_per_item(self) -> pd.Series:
+    def length_per_item(self) -> pd.Series:
```

Comment:
How is this making `length` consistent with other naming? Do we have any other APIs where we refer to "length" of timeseries? I think `num_timesteps` is more explicit---`length` in the context of time may refer to absolute time (2 days, etc.) or python `__len__`.

Reply:
My idea was that if we remove `timestep` from the slice method, then we can also remove it here since `length` sounds a bit more intuitive, coincides with python `__len__` for each item, and is more in line with how we usually use this information in other methods
```python
for item_id, length in df.length_per_item().iteritems():
    ...
```
vs.
```python
for item_id, length in df.num_timesteps_per_item().iteritems():
    ...
```
But I'm also fine with keeping it as `num_timesteps_per_item`, not really a strong preference.

Reply:
`num_timesteps_per_item` then?